pkgs/development/python-modules/ifcopenshell/default.nix

```diff
@@ -1,62 +1,178 @@
 {
   lib,
   stdenv,
+  testers,
   buildPythonPackage,
+  pythonOlder,
+  # fetchers
   fetchFromGitHub,
-  gcc10,
+  # build tools
   cmake,
+  gcc10,
+  python,
+  swig,
+  # native dependencies
+  eigen,
   boost179,
+  cgal,
+  gmp,
+  hdf5,
   icu,
-  swig,
-  pcre,
+  libaec,
+  libxml2,
+  mpfr,
+  nlohmann_json,
   opencascade-occt_7_6,
   opencollada,
-  libxml2,
+  pcre,
+  zlib,
+
+  # python deps
+  ## tools
+  setuptools,
+  build,
+  wheel,
+  pytest,
+  ## dependencies
+  isodate,
+  lark,
+  numpy,
+  python-dateutil,
+  shapely,
+  typing-extensions,
+  ## additional deps for tests
+  lxml,
+  mathutils,
+  networkx,
+  tabulate,
+  xmlschema,
+  xsdata,
 }:
 let
   opencascade-occt = opencascade-occt_7_6;
 in
 buildPythonPackage rec {
   pname = "ifcopenshell";
-  version = "240611";
+  version = "0.7.10";
   format = "other";
+  disabled = pythonOlder "3.9";
 
   src = fetchFromGitHub {
     owner = "IfcOpenShell";
     repo = "IfcOpenShell";
-    rev = "refs/tags/blenderbim-${version}";
+    rev = "refs/tags/ifcopenshell-python-${version}";
     fetchSubmodules = true;
-    hash = "sha256-fCgxSambC01qb14EoZTrVxNNw8H3nX8oMi6gk0O/HmQ=";
+    hash = "sha256-cRzv07T5VN5aTjMtAlLGbvI3c4SL0lfzCn/W6f/vdBY=";
   };
 
   nativeBuildInputs = [
-    gcc10
+    # c++
     cmake
+    gcc10
```

Comment:
Where is the version requirement mentioned/ Why is it needed? Please add a comment.

Reply:
No it's not, I removed it.